x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -365,7 +364,7 @@ public void restoreShard(
             ),
             () -> IOUtils.close(toClose)
         );
-        try {
+        ActionListener.run(restoreListener, restoreListener1 -> {
```

Comment:
Sorry should have clarified - you can reasonably inline the definition of `restoreListener` here and then rename `restoreListener1` to `restoreListener`.